web/src/main/java/org/springframework/security/web/session/SessionLimitStrategy.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.session;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.util.Assert;
+
+/**
+ * Represents the maximum number of sessions allowed. Use {@link #UNLIMITED} to indicate
+ * that there is no limit.
+ *
+ * @author Claudenir Freitas
+ * @since 6.5
+ */
+@FunctionalInterface
+public interface SessionLimitStrategy {
+
+	int resolve(Authentication authentication);
+
+	/**
+	 * Represents unlimited sessions.
+	 */
+	SessionLimitStrategy UNLIMITED = (authentication) -> -1;
+
+	/**
+	 * Creates a {@link SessionLimitStrategy} that always returns the given value for any
+	 * user
+	 * @param maxSessions the maximum number of sessions allowed
+	 * @return a {@link SessionLimitStrategy} instance that returns the given value.
+	 */
+	static SessionLimitStrategy of(int maxSessions) {
+		Assert.isTrue(maxSessions != 0 && maxSessions >= -1,
```

Comment:
I think `maxSessions >  0` will simplify this logic, or am I missing something?

Reply:
I see. I would prefer that we abstract away sentinel values. Instead, they should use something like `SessionLimit.UNLIMITED` to indicate that they want unlimited sessions. `of` should only take values greater than 0.